### PR TITLE
swap out old DRP documentation with v.0.3 draft

### PR DESCRIPTION
**Pull request by rrix**
i wonder how I can get this thing to render just the markdown to comment on...

Here's my spin at a tight 0.3 of just the technical bits and data schemas.

### CODE REVIEW

File: data-rights-protocol.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+Version 0.3 encodes the provisions of the California Code bla bla section bla as specified in the California Consumer Privacy act of 2018, referred herein as the “CCPA” and intends to encode the following rights. This is further enumerated in the [Supported Rights Actions]() section of this document below.
+
+### 0.03 Terminology
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “NOT RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in BCP 14 [RFC2119] [RFC8174] when, and only when, they appear in all capitals, as shown here.
+
+* A **Data Request** is a JSON object which encodes the rights and identity of an End-User, identified by a globally unique identifier.
+* **Requester** is the End-User or Authorized Agent who communicates with the **Request Processor** over the DRP protocol.
+* **Request Processor** is the entity that processes a Data Request, either a Covered Business directly or a **Privacy Infrastructure Provider**.
```
Comment (rrix) [inline data-rights-protocol.md:25]: @dazzaji we should put these in the same place as your glossary and include them by-reference to that.

File: data-rights-protocol.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+## 2.0 HTTP Endpoint Specification
+
+[note about including schemas by-reference from below.]
```
Comment (rrix) [inline data-rights-protocol.md:29]: add

File: data-rights-protocol.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+
+An End-User looking to exercise their data rights for Example, Inc. whose homepage is https://example.com/, for example, MUST be able to GET https://example.com/.well-known/data-rights without knowledge of the Covered Business’s relationship to any Privacy Infrastructure Provider. DSR Providers SHOULD provide a mechanism to prepare and process these endpoint requests but the Covered Business is responsible for receiving this request.
+
+This `well-known` resource SHOULD be registered with IANA before 1.0 specification.
+
+```
+{
+  "version": "0.3",
+  "api_base": "https://dsr-provider.example.com/data-rights",
+  "actions": ["do_not_sell", "access", "deletion"],
```
Comment (rrix) [inline data-rights-protocol.md:48]: update actions for

- `sale:opt_in`
- `sale:opt_out`

File: data-rights-protocol.md [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+This is the Data Rights Exercise endpoint which End-Users and Authorized Agents can use to exercise enumerated data rights.
+
+```
+{
+  "meta": {
+    "version": "0.3"
+  },
+  "regulatory_authority": "ccpa",
+  "exercise": [
+    "do_not_sell"
```
Comment (rrix) [inline data-rights-protocol.md:67]: references old right enumerator

File: data-rights-protocol.md [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+- `meta` MUST contain only a single key `version` which contains a string referencing the current protocol version “0.3”.
+- `regulatory_authority` MUST contain a string referencing the regulation under which the Data Request is being taken under
+- `exercise` MUST contain a list of rights to exercise.
+- `identity` MUST contain an [RFC7515 JWT](https://datatracker.ietf.org/doc/html/rfc7515) conforming to one of the following specifications:
+  - a string containing a JWT serialized in the Compact Serialization format [RFC7515 Section 3.1]
+  - a document object containing a JWT serialized in the JSON Serialization formation [RFC7515 Section 3.2]
+- `status_callback` MAY be specified with a URL that the Status Callback can be sent to.
+
+See section 3.04 regarding identity encapsulation.
```
Comment (rrix) [inline data-rights-protocol.md:82]: urlize

File: data-rights-protocol.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+Covered Business SHOULD make a best effort to ensure that a 200 status is recorded for the most recent status update. The body of the callback's response SHOULD be treated as opaque and not be considered for parsing by the Covered Business.
+
+### 2.05 `POST /cancel` ("Data Rights Cancel" endpoint)
+
+[we still need to reason about this, there are some open questiosn I haven't spent much time thinking about on how to line up "revoking consent" with "canceling request" between DNS and access/deletion]
+
+```
+{
+  "request_id": "c789ff35-7644-4ceb-9981-4b35c264aac3",
+  "reason": "i don't want my account deleted"
```
Comment (rrix) [inline data-rights-protocol.md:114]: this is in a grab-bag of "nice to have" optional fields that i had in mind. feel free to redline

File: data-rights-protocol.md [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+
+* canceling an access request is “easy”, “just” terminate internal export pipeline
+* canceling a DNS consent revocation might be meaningless, might just be a database column update?
+* canceling a deletion request is a bit more complicated: if the process has already started there’s no real situation where it’s “safe” to cancel it. In general
+    * experience here tells me that deletion should not be immediately enacted, covered businesses SHOULD schedule deletion 14 or 30 days or so to provide consumers a chance to cancel the request.
+    * unless deletion-scope narrowing is in play here I think we should assume account-level deletion will occur along with any BoH deletion?
+
+## 3.0 Data Schemas
+
+These Schemas are referenced in Section 3 outlining the HTTP endpoints and their semantics.
```
Comment (rrix) [inline data-rights-protocol.md:136]: change number

File: data-rights-protocol.md [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+| `address`                                 | if known, claim SHALL contain the user's preferred address, if known.                                                                                                                                       |
+
+Covered Businesses SHALL make the final determination over whether a given token is trustworthy. Agents SHALL make a best effort to provide trustworthy tokens, by attaching user-attested claims as available, and by ensuring their JWTs are signed by a key which the Covered Businesses and PIPs can verify against. 
+
+JWTs should probably not be encrypted? managing the encryption key exchange here is very strange and necessarily happening out-of-scope of the protocol. but we already have shared-secret API authentication in section 3.07; I am queasy about having user tokens identity floating in the open here....
+
+### 3.06 Error States
+
+Todo, error states in processing, error states in POST exercise, etc...
```
Comment (rrix) [inline data-rights-protocol.md:220]: still punting on thinking about this, really need to spend some time considering it more before unveiling.

**Comment by rrix** [comment]
I really wish I could signal to @GitHub that I don't care that it doesn't render "large diffs" by default, I need this doc reviewed and it's a UX pain.

**Comment by dazzaji** [comment]
This is great!

**Pull request merged**